### ISSUE TITLE
Generic.Arrays.ArrayIndent and Generic.WhiteSpace.ScopeIndent conflict

### ISSUE THREAD

**Issue by geerteltink** [squizlabs/PHP_CodeSniffer#2905]
#### Summary

phpcbf loops until attempts limit (50) with rules Generic.Arrays.ArrayIndent & Generic.WhiteSpace.ScopeIndent then failed to fix.
On big sources failed on time limit (10min)

#### Current behavior

```
$./vendor/bin/phpcbf test.php
----------------------------------------------------------------------
FILE                                                  FIXED  REMAINING
----------------------------------------------------------------------
test.php                                              FAILED TO FIX
----------------------------------------------------------------------
A TOTAL OF 1 ERROR WERE FIXED IN 1 FILE
----------------------------------------------------------------------
PHPCBF FAILED TO FIX 1 FILE
----------------------------------------------------------------------
```

#### How to reproduce

Using the defaults for Generic.Arrays.ArrayIndent and Generic.WhiteSpace.ScopeIndent.

Minimal source to reproduce:
```php
<?php

function test()
{
    query('
        INSERT INTO table (column)
        VALUES (:value)
    ', [
        'value' => $value,
    ]);
}
```

#### Expected behavior

No error (all other rules applied)

#### Verbose output fragment

```
$./vendor/bin/phpcbf test.php -vvvv

---START FILE CONTENT---
 1|<?php
 2|
 3|function test()
 4|{
 5|    query('
 6|        INSERT INTO table (column)
 7|        VALUES (:value)
 8|    ', [
 9|    'value' => $value,
10|    ]);
11|}
12|
--- END FILE CONTENT ---
	**** Generic.Arrays.ArrayIndent:137 has possible conflict with another sniff on loop 45; caused by the following change ****
	**** replaced token 29 (T_WHITESPACE on line 10) "····]" => "]" ****
	**** ignoring all changes until next loop ****
        => Fixing file: 0/2 violations remaining [made 47 passes]...    * fixed 0 violations, starting loop 48 *
---START FILE CONTENT---
 1|<?php
 2|
 3|function test()
 4|{
 5|    query('
 6|        INSERT INTO table (column)
 7|        VALUES (:value)
 8|    ', [
 9|    'value' => $value,
10|    ]);
11|}
12|
--- END FILE CONTENT ---
	Generic.Arrays.ArrayIndent:137 replaced token 29 (T_WHITESPACE on line 10) "····]" => "]"
        => Fixing file: 1/2 violations remaining [made 48 passes]...    * fixed 1 violations, starting loop 49 *
---START FILE CONTENT---
 1|<?php
 2|
 3|function test()
 4|{
 5|    query('
 6|        INSERT INTO table (column)
 7|        VALUES (:value)
 8|    ', [
 9|    'value' => $value,
10|]);
11|}
12|
--- END FILE CONTENT ---
	**** Generic.WhiteSpace.ScopeIndent:1476 has possible conflict with another sniff on loop 47; caused by the following change ****
	**** replaced token 29 (T_CLOSE_SHORT_ARRAY on line 10) "]" => "····]" ****
	**** ignoring all changes until next loop ****
        => Fixing file: 0/2 violations remaining [made 49 passes]...    * fixed 0 violations, starting loop 50 *
---START FILE CONTENT---
 1|<?php
 2|
 3|function test()
 4|{
 5|    query('
 6|        INSERT INTO table (column)
 7|        VALUES (:value)
 8|    ', [
 9|    'value' => $value,
10|]);
11|}
12|
--- END FILE CONTENT ---
	Generic.WhiteSpace.ScopeIndent:1476 replaced token 29 (T_CLOSE_SHORT_ARRAY on line 10) "]" => "····]"
        => Fixing file: 1/2 violations remaining [made 50 passes]...    * fixed 1 violations, starting loop 51 *
	*** Reached maximum number of loops with 1 violations left unfixed ***
ERROR in 217ms

PHPCBF RESULT SUMMARY
----------------------------------------------------------------------
FILE                                                  FIXED  REMAINING
----------------------------------------------------------------------
test.php                                              FAILED TO FIX
----------------------------------------------------------------------
A TOTAL OF 1 ERROR WERE FIXED IN 1 FILE
----------------------------------------------------------------------
PHPCBF FAILED TO FIX 1 FILE
----------------------------------------------------------------------

Time: 292ms; Memory: 10MB
```

**Comment by gsherwood** [comment]
I don't get a conflict when I run those 2 sniffs over the sample code provided, but I think something is still broken.

I get this:
```
$ phpcs temp.php --sniffs=Generic.Arrays.ArrayIndent,Generic.WhiteSpace.ScopeIndent --report=diff -vvv
...
	*** START FILE FIXING ***
---START FILE CONTENT---
 1|<?php
 2|
 3|function test()
 4|{
 5|    query('
 6|        INSERT INTO table (column)
 7|        VALUES (:value)
 8|    ', [
 9|        'value' => $value,
10|    ]);
11|}
12|
--- END FILE CONTENT ---
	E: [Line 9] Array key not indented correctly; expected 4 spaces but found 8 (Generic.Arrays.ArrayIndent.KeyIncorrect)
	Generic.Arrays.ArrayIndent:100 replaced token 21 (T_WHITESPACE on line 9) "········'value'" => "····'value'"
	E: [Line 10] Array close brace not indented correctly; expected 0 spaces but found 4 (Generic.Arrays.ArrayIndent.CloseBraceIncorrect)
	Generic.Arrays.ArrayIndent:137 replaced token 29 (T_WHITESPACE on line 10) "····]" => "]"
	* fixed 2 violations, starting loop 2 *
---START FILE CONTENT---
 1|<?php
 2|
 3|function test()
 4|{
 5|    query('
 6|        INSERT INTO table (column)
 7|        VALUES (:value)
 8|    ', [
 9|    'value' => $value,
10|]);
11|}
12|
--- END FILE CONTENT ---
	*** END FILE FIXING ***
--- temp.php
+++ PHP_CodeSniffer
@@ -6,6 +6,6 @@
         INSERT INTO table (column)
         VALUES (:value)
     ', [
-        'value' => $value,
-    ]);
+    'value' => $value,
+]);
 }
```
But it looks like the errors from `Generic.Array.ArrayIndent` are wrong. Maybe fixing those will fix your issue, but I can't reproduce to confirm that. If you have any more information about how to reproduce, please let me know.
